### ISSUE TITLE
ejsLog() kills the erlang vm in a reduce

### ISSUE THREAD

**Issue by clr** [basho/riak#209]
If you uncomment the following line:
https://github.com/clr/riak-rolling-average/blob/mapreduce/lib/statistic_document.rb#L37
Then the entire erlang vm disappears.  No log files are marked.  I tested can repeat this on memory backend, and Randy Secrist can repeat on leveldb.  Riak v1.2.0.

**Comment by staffan-einarsson** [comment]
I can repeat this as well in my tests. It seems from dmesg that using the ejsLog function in a mapreduce phase causes a memory protection fault in the Erlang JavaScript driver. I got the following line:

<pre>[26160.090712] beam[3822] general protection ip:7f891c1ef5cb sp:7f891d6eba90 error:0 in erlang_js_drv.so[7f891c14f000+e7000]</pre>

EDIT:
It happens if I use ejsLog() from either the map phase or reduce phase. So it's not isolated to reduce.

Here are two phase functions that I use that can reproduce the problem. Uncommenting either of the two ejsLog calls will crash the Erlang VM.

My Map function is:

```
function(object, keydata, arg)
{
    //ejsLog('mapreduce.log', 'Hello!');
    return [JSON.parse(object.values[0].data)];
}
```

My Reduce function is:

```
function(valueList, keydata, arg)
{
    //ejsLog('mapreduce.log', 'Hello!');
    valueList.sort();
    var result = valueList;
    return result;
}
```

In this case the MapReduce was sent to operate on a whole bucket.

EDIT 2:

I'm running Riak 1.2.0 on Ubuntu Server 12.04.

**Comment by shuveb** [comment]
Yes, I can confirm the same on MacOS X Lion. The "beam.smp" process simply crashes. Logs tell me it is: EXC_BAD_ACCESS (SIGSEGV)

If I comment the ejsLog() function, things seem to be just fine.

Riak version I'm running is 1.2.0

**Comment by dizzyd** [comment]
Fixed in https://github.com/basho/erlang_js/pull/30

**Comment by dizzyd** [comment]
Merged.

**Comment by jaredmorrow** [comment]
Slated for Riak 1.2.1 release.

**Comment by stoddn** [comment]
Verified fix in 1.2.1rc2 on CentOS 5.6 with LevelDB.

**Comment by jaredmorrow** [comment]
Thanks @stoddn
